src/perception/long_range_cam/long_range_cam.cpp

```diff
@@ -0,0 +1,86 @@
+#include "long_range_cam.hpp"
+#include <opencv2/imgproc.hpp>
+#include <opencv2/videoio.hpp>
+#include <ros/init.h>
+#include <ros/this_node.h>
+#include <ros/time.h>
+#include <sensor_msgs/image_encodings.h>
+#include <thread>
+
+namespace mrover {
+
+    using namespace std::chrono_literals;
+
+    /**
+    * @brief Load config, open the camera, and start our threads
+    */
+    void LongRangeCamNodelet::onInit() {
+        try {
+            mGrabThread = std::jthread(&LongRangeCamNodelet::grabUpdate, this);
+        } catch (std::exception const& e) {
+            NODELET_FATAL("Exception while starting: %s", e.what());
+            ros::shutdown();
+        }
+    }
+
+    void fillImageMessage(cv::Mat& bgra, sensor_msgs::ImagePtr const& msg) {
+        assert(msg);
```

Comment:
I would add a couple of asserts. You need to make sure that `bgra` is continous (i.e. it is not a slice, bc then you can't just memcpy), and more importantly check the opencv mat type

Reply:
Also, shouldn't bgra be renamed to bgr?